eventauth.go

```diff
@@ -487,7 +486,7 @@ func (a *allowerContext) aliasEventAllowed(event PDU) error {
 	// Check that event is a state event.
 	// Check that the state key matches the server sending this event.
 	// https://github.com/matrix-org/synapse/blob/v0.18.5/synapse/api/auth.py#L158
-	if !event.StateKeyEquals(string(sender.Domain())) {
+	if !event.StateKeyEquals(string(event.SenderID())) {
```

Comment:
This isn't correct for normal Matrix versions which have the state key set to the domain of the HS e.g `state_key: matrix.org`.